### PR TITLE
Ensure sub-query use for Job instances.

### PR DESCRIPTION
**Pull request by corneil**
Fix find by name query ordering.
 Fix find by name not found query.
 Improve test case readability.
 #5484

### CODE REVIEW

**Review by onobc — CHANGES_REQUESTED** [review]
Thanks for the PR @corneil . I have a few comments/change requests to take a look at please. 

Also, is there a test that exhibits the bug this is fixing? If not, can we make one or is this a performance related issue?

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java
Comment (cppwfs) [reply spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java:122]: They are good for diagnostic, but once we have diagnosed the issue, we should remove them.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java:129]: We can remove the print here and below.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java
Comment (onobc) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:189]: Understood. Thanks for fixing it.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java:55]: Missed this wildcard.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java
Comment (onobc) [reply spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java:126]: Discussed offline.

**Review by onobc — APPROVED** [review]
@corneil LGTM - thanks for the updates. There are still a couple of wildcard imports. Once these are adjusted consider this PR approved. I am approving now but please update the imports.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcAggregateJobQueryDao.java
Comment (onobc) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcAggregateJobQueryDao.java]: @corneil please answer this one.

Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcAggregateJobQueryDao.java]: The ternary operator isn't needed and refactored.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java]: done

**Pull request merged**
